python/src/uagents/context.py

```diff
@@ -49,275 +54,277 @@
+            message_body (JsonStr): The JSON-encoded message body to be sent.
+            sync (bool): Whether to send the message synchronously or asynchronously.
+            timeout (Optional[int]): The optional timeout for sending the message, in seconds.
+            protocol_digest (Optional[str]): The protocol digest of the message to be sent.
 
         Returns:
-            Dict[str, Tuple[JsonStr, str]]: The dictionary of outbound messages.
+            MsgStatus: The delivery status of the message.
         """
-        return self._outbound_messages
+        pass
 
-    def get_message_protocol(self, message_schema_digest) -> Optional[str]:
+    @abstractmethod
+    async def send_wallet_message(
+        self,
+        destination: str,
+        text: str,
+        msg_type: int = 1,
+    ):
         """
-        Get the protocol digest associated with a given message schema digest.
+        Send a message to the wallet of the specified destination.
 
         Args:
-            message_schema_digest (str): The schema digest of the message.
+            destination (str): The destination address to send the message to.
+            text (str): The text of the message to be sent.
+            msg_type (int): The type of the message to be sent.
 
         Returns:
-            Optional[str]: The protocol digest associated with the message schema digest,
-            or None if not found.
+            None
         """
-        for protocol_digest, protocol in self._protocols.items():
-            for model in protocol.models:
-                if message_schema_digest == model:
-                    return protocol_digest
-        return None
+        pass
+
+
+class InternalContext(Context):
+    """
+    Represents the agent internal context for proactive behaviour.
+    """
+
+    def __init__(
+        self,
+        agent: AgentRepresentation,
+        storage: KeyValueStore,
+        ledger: LedgerClient,
+        resolver: Resolver,
+        dispenser: Dispenser,
+        interval_messages: Optional[Set[str]] = None,
+        wallet_messaging_client: Optional[Any] = None,
+        logger: Optional[logging.Logger] = None,
+    ):
+        self._agent = agent
+        self._storage = storage
+        self._ledger = ledger
+        self._resolver = resolver
+        self._dispenser = dispenser
+        self._logger = logger
+        self._session: Optional[uuid.UUID] = None
+        self._interval_messages = interval_messages
+        self._wallet_messaging_client = wallet_messaging_client
+        self._outbound_messages: Dict[str, Tuple[JsonStr, str]] = {}
+
+    @property
+    def agent(self) -> AgentRepresentation:
+        return self._agent
+
+    @property
+    def storage(self) -> KeyValueStore:
+        return self._storage
+
+    @property
+    def ledger(self) -> LedgerClient:
+        return self._ledger
 
-    def update_protocols(self, protocol: Protocol) -> None:
```

Comment:
just to double check: Dialogues are still working with this removed?

Reply:
Yes, the Dialogue examples have been thoroughly tested with the new context and this was only needed at the stage where Dialogues were more deeply embedded in the context itself.